src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/create_interaction_choice_set_request.cc

```diff
@@ -462,6 +467,14 @@ void CreateInteractionChoiceSetRequest::DeleteChoices() {
 void CreateInteractionChoiceSetRequest::OnAllHMIResponsesReceived() {
   SDL_LOG_AUTO_TRACE();
 
+  ApplicationSharedPtr application =
+      application_manager_.application(connection_key());
+  if (!error_from_hmi_) {
+    application->set_choice_set_allow_mode(choice_set_id_, true);
```

Comment:
@ychernysheva don't we need to call `set_choice_set_allow_mode` in `DeleteChoices()` ?

Reply:
@AKalinich-Luxoft I think no, because now we use std::set to store allowed choice set ids. We mark choice set as allowed when it is fully processed. In this case processing was interrupted by error response, and we hadn't added corresponding id to std::set before, so there is no id to delete.